Marlin/src/libs/softspi.h

```diff
@@ -75,10 +75,7 @@
      * @param[in] level value to write
      */
     FORCE_INLINE static void fastDigitalWrite(uint8_t pin, bool value) {
-      if (value)
-        g_APinDescription[pin].pPort->PIO_SODR = g_APinDescription[pin].ulPin;
-      else
-        g_APinDescription[pin].pPort->PIO_CODR = g_APinDescription[pin].ulPin;
+      digitalWrite(pin, value);
```

Comment:
i think that require a cast to pin_t(pin) to use the HAL digitalWrite()

Reply:
Marlin/src/HAL/HAL_STM32F1/HAL.h:201:typedef int8_t pin_t;
Marlin/src/HAL/HAL_STM32/HAL.h:153:typedef int16_t pin_t;

Reply:
but maybe i make a confusion with analogWrite... i just see this one
Marlin/src/HAL/HAL_LINUX/include/Arduino.h:104:void digitalWrite(pin_t, uint8_t);

Reply:
`digitalWrite` is of course the Arduino "slow" pin write routine, which may be fully optimized on some platforms, but on AVR we bypass it and use our FastIO `WRITE` routine whenever possible. Since this is a `libs` entry it should only use the fastest pin manipulation technique available in the headers (e.g., `<Arduino.h>`) it depends on.